scholarx/archive/functions.js

```diff
@@ -78,19 +82,41 @@ function renderAllProfiles() {
     $("#mentorProfiles").html(mentorProfiles);
     $("#menteeProfiles").html(menteeProfiles);
 }
+function renderCohortCheckboxes(){
+    const checkboxYears = []
+    for(let i=0; i<years.length; i++){
+        checkboxYears.push({ id: years[i] });
+    }
+    var data = {checkboxes:checkboxYears};
+    var template = document.getElementById("cohort").innerHTML;
+    var output = Mustache.render(template, data);
+    document.getElementById("cohort-filters").innerHTML = output;
+}
+function checkboxCheckStatus(){
+    for(let i=years[0]; i<=years[years.length-1]; i++){  //function to return true if all checkboxes are not selected
+        if(document.getElementById(i).checked){
+            return false;
+        }
+    }return true;
+}
 function filterByYear(){
     let mentorsData = [];
     let menteesData = [];
-    if(!document.getElementById("2019").checked && !document.getElementById("2020").checked && !document.getElementById("2021").checked){
+    if(checkboxCheckStatus()){
         renderAllProfiles();
     }else{
-        for(let year=2019; year<2022; year++){
+        for(let year=years[0]; year<=years[years.length-1]; year++){
             if(document.getElementById(year).checked)
             {
-                mentorsData = mentorsData.concat(mentors[year]),
-                menteesData = menteesData.concat(mentees[year]),
-                renderProfiles(mentorsData,menteesData)
+                function checkYears(element) {
+                    return element.year == year;
+                }
+                const filteredMentors = mentors.filter(checkYears);
+                const filteredMentees = mentees.filter(checkYears);
+                mentorsData = mentorsData.concat(filteredMentors);
+                menteesData = menteesData.concat(filteredMentees);
             }
         }
```

Comment:
```js
function filterByYear() {
   // You'll have to write getSelectedYears function elsewhere
   const selectedYears = getSelectedYears();
   if (selectedYears.length == 0 || selectedYears.length == years.length) {
      renderAllProfiles();
      return;
   }
   filteredMentors = mentors.filter((mentor) => selectedYears.includes(mentor.year));
   filteredMentees = mentees.filter((mentee) => selectedYears.includes(mentee.year));
   renderProfiles(filteredMentors, filteredMentees);
}
```

Reply:
The previous method loops through the entire mentor array `years.length` times, each time a checkbox is checked or unchecked. Which is not the most optimal way of doing things. This would slightly increase the efficiency of the code. 

Also we can safely remove `checkCheckboxStatus` function.